### PR TITLE
Feature/status notification

### PR DESCRIPTION
**Pull request by matteopariset**
Display a notification when:
1. The UI is not running but the app is running in the background
2. Internet connection is not active

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/connectivity/ConcreteConnectivityBroker.java [hunk 1]
```diff
@@ -33,23 +35,58 @@
...
+    }
+
     private ConnectivityManager.NetworkCallback internetNetworkCallback = new ConnectivityManager.NetworkCallback() {
+
         @Override
         public void onCapabilitiesChanged(@NonNull Network network, @NonNull NetworkCapabilities networkCapabilities) {
             super.onCapabilitiesChanged(network, networkCapabilities);
-            if (networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)) {
+            if (checkInternetConnection()) {
+                Log.e("BROKER", "Internet capabilities changed");
```
Comment (TheBlueHawk) [inline app/src/main/java/ch/epfl/sdp/connectivity/ConcreteConnectivityBroker.java]: Should we keep this log?

File: app/src/main/res/values-fr-rFR/strings.xml [hunk 2]
```diff
@@ -3,7 +3,7 @@
     <string name="accountDetails">Détails du compte</string>
 
     <string name="aggregator_status_off">Zut! Vous êtes hors ligne</string>
-    <string name="aggregator_status_on">Vous êtes maintenant en ligne</string>
+    <string name="aggregator_status_on">Vous êtes maintenant protégés</string>
```
Comment (TheCrazyKing) [inline app/src/main/res/values-fr-rFR/strings.xml]: pas de 's' à prétégé, il s'agit d'une seule personne (forme de politesse)

File: app/src/main/java/ch/epfl/sdp/connectivity/ConcreteConnectivityBroker.java [hunk 3]
```diff
@@ -33,23 +35,58 @@
...
+                setChanged();
+                notifyObservers(true);
+            }
+        }
+
+        @Override
+        public void onAvailable(@NonNull Network network) {
+            super.onAvailable(network);
+            if (checkInternetConnection()) {
+                Log.e("BROKER", "Internet available");
```
Comment (TheBlueHawk) [inline app/src/main/java/ch/epfl/sdp/connectivity/ConcreteConnectivityBroker.java]: Same as before

File: app/src/main/java/ch/epfl/sdp/contamination/ConcretePositionAggregator.java [hunk 4]
```diff
@@ -152,4 +152,9 @@ public void updateToOffline() {
     public void updateToOnline() {
         this.isOnline = true;
     }
+
+    @Override
+    public void stopAggregator() {
+        updatePosTimer.cancel();
```
Comment (TheCrazyKing) [inline app/src/main/java/ch/epfl/sdp/contamination/ConcretePositionAggregator.java]: Why would one call it ? And when does it start again ?
As I understand, we call stopAggregator() when you have no GPS connection, but where does it start again ?

**Review by TheCrazyKing — CHANGES_REQUESTED** [review]
Nice feature. I'have little quesionts though.

File: app/src/main/res/values-fr-rFR/strings.xml [hunk 5]
```diff
@@ -123,10 +123,11 @@
     <string name="missing_GPS_Perm_Dialog_title">autorisation GPS manquante</string>
     <string name="missing_GPS_Perm_Dialog_text">Vous avez refusé l\'autorisation GPS. L\'application
     en a besoin pour fonctionner. Cliquez OUI puis autorisez ou cliquez annuler pour fermer l\'application</string>
-
+    <string name="internet_offline_msg">Vous êtes hors ligne</string>
+    <string name="background_protection_msg">Votre protection est encore activée</string>
```
Comment (TheBlueHawk) [inline app/src/main/res/values-fr-rFR/strings.xml:127]: Isn't a bit misleading saying that the app is "protecting" the user?

File: app/src/main/java/ch/epfl/sdp/location/LocationService.java [hunk 6]
```diff
@@ -223,16 +313,34 @@ public int onStartCommand(Intent intent, int flags, int startId) {
     @Override
     public void update(Observable o, Object arg) {
         // Store updates to Carrier
-        locallyStoreCarrier();
-        AsyncTask.execute(() -> remotelyStoreCarrierStatus(analyst.getCarrier().getInfectionStatus()));
+        AsyncTask.execute(() -> {
```
Comment (AdrienRem) [inline app/src/main/java/ch/epfl/sdp/location/LocationService.java:316]: Are you sure that you don't interact with UI components in theses functions ?

You have to declare a subclass as explained here if that's the case or you may run into some nasty concurrency issues: https://developer.android.com/reference/android/os/AsyncTask

**Review by TheBlueHawk — CHANGES_REQUESTED** [review]
Good work, but shouldn't we remove all these log before publishing the app? And why the french traduction has a different meaning?

File: app/src/main/res/values-fr-rFR/strings.xml (hunk 5 above)
Comment (matteopariset) [reply app/src/main/res/values-fr-rFR/strings.xml:127]: Agree, just marketing

File: app/src/main/java/ch/epfl/sdp/connectivity/ConcreteConnectivityBroker.java (hunk 1 above)
Comment (matteopariset) [reply app/src/main/java/ch/epfl/sdp/connectivity/ConcreteConnectivityBroker.java]: It was useful to check which one of the two methods was called

File: app/src/main/java/ch/epfl/sdp/connectivity/ConcreteConnectivityBroker.java (hunk 3 above)
Comment (matteopariset) [reply app/src/main/java/ch/epfl/sdp/connectivity/ConcreteConnectivityBroker.java]: 🔼

File: app/src/main/java/ch/epfl/sdp/location/LocationService.java (hunk 6 above)
Comment (matteopariset) [reply app/src/main/java/ch/epfl/sdp/location/LocationService.java:316]: I'm quite sure

File: app/src/main/java/ch/epfl/sdp/contamination/ConcretePositionAggregator.java (hunk 4 above)
Comment (matteopariset) [reply app/src/main/java/ch/epfl/sdp/contamination/ConcretePositionAggregator.java]: Ohh, leftover function

File: app/src/main/res/values-fr-rFR/strings.xml (hunk 2 above)
Comment (matteopariset) [reply app/src/main/res/values-fr-rFR/strings.xml]: Je modifie

File: app/src/main/java/ch/epfl/sdp/contamination/ConcretePositionAggregator.java (hunk 4 above)
Comment (TheCrazyKing) [reply app/src/main/java/ch/epfl/sdp/contamination/ConcretePositionAggregator.java]: as you said

File: app/src/main/java/ch/epfl/sdp/location/LocationService.java (hunk 6 above)
Comment (AdrienRem) [reply app/src/main/java/ch/epfl/sdp/location/LocationService.java:316]: I'm quite convinced then

**Review by TheCrazyKing — APPROVED** [review]

**Review by TheBlueHawk — APPROVED** [review]

**Pull request merged**